mongo/dataset_store.go

```diff
@@ -16,6 +17,31 @@ import (
 	bsonprim "go.mongodb.org/mongo-driver/bson/primitive"
 )
 
+// GetDatasetsByBasedOn checks Published...... TODO: FINISH
+// Filter condition checks for ... .TODO: FINISH
+func (m *Mongo) GetDatasetsByBasedOn(ctx context.Context, id string, offset, limit int, authorised bool) (values []*models.DatasetUpdate, totalCount int, err error) {
+	var filter = bson.M{
+		"$or": bson.A{
+			bson.M{"current.is_based_on.id": id},
+			bson.M{"next.is_based_on.id": id},
+		},
+	}
+
+	values = []*models.DatasetUpdate{}
+	totalCount, err = m.Connection.Collection(m.ActualCollectionName(config.DatasetsCollection)).Find(ctx, filter, &values,
```

Comment:
You can definitely format this line better. As a rule of thumb, when breaking the line during a function call you should do so immediately rather than arbitrarily. The exception is if you're opening new scope within the call. e.g.

```
Stuff.Do(
  arg1,
  arg2,
)
```
or
```
Stuff.Do(arg1, arg2, arg3{
  field1,
  field2,
})
```
In combination you might want to use the newline-after-function-call here too if the lines are still too long.
Maybe something like this:
```
	totalCount, err = m.Connection.
		Collection(m.ActualCollectionName(config.DatasetsCollection)).
		Find(
			ctx,
			filter,
			&values,
			mongodriver.Sort(bson.M{"_id": -1}),
			mongodriver.Offset(offset),
			mongodriver.Limit(limit),
		)
```